app/src/main/java/com/x8bit/bitwarden/data/platform/repository/BridgeRepositoryImpl.kt

```diff
@@ -0,0 +1,117 @@
+            }
+
+            // Unlock vault if necessary:
+            val isVaultAlreadyUnlocked = vaultRepository.isVaultUnlocked(userId = userId)
+            if (!isVaultAlreadyUnlocked) {
+                val unlockResult = vaultRepository
+                    .unlockVaultWithDecryptedUserKey(
+                        userId = userId,
+                        decryptedUserKey = decryptedUserKey,
+                    )
+
+                when (unlockResult) {
+                    is VaultUnlockResult.AuthenticationError,
+                    VaultUnlockResult.GenericError,
+                    VaultUnlockResult.InvalidStateError,
+                    -> {
+                        // Not being able to unlock the user's vault with the
+                        // decrypted unlock key is an unexpected case, but if it does
+                        // happen we omit the account from list of shared accounts
+                        // and remove that user's authenticator sync unlock key.
+                        // This gives the user a way to potentially re-enable syncing
+                        // (going to Account Security and re-enabling the toggle)
+                        authDiskSource.storeAuthenticatorSyncUnlockKey(
+                            userId = userId,
+                            authenticatorSyncUnlockKey = null,
+                        )
+                        return@mapNotNull null
+                    }
+                    // Proceed
+                    VaultUnlockResult.Success -> Unit
+                }
+            }
+
+            // Vault is unlocked, query vault disk source for totp logins:
+            val totpUris = vaultDiskSource
+                .getCiphers(userId)
+                .first()
+                // Filter out any ciphers without a totp item:
+                .filter { it.login?.totp != null }
+                .mapNotNull {
+                    // Decrypt each cipher and take just totp codes:
+                    vaultSdkSource
+                        .decryptCipher(
+                            userId = userId,
+                            cipher = it.toEncryptedSdkCipher(),
+                        )
+                        .getOrNull()
+                        ?.login
+                        ?.totp
+                }
+
+            val lastSyncTime =
+                settingsDiskSource.getLastSyncTime(userId) ?: return@mapNotNull null
+
+            // Lock the user's vault if we unlocked it for this operation:
+            if (!isVaultAlreadyUnlocked) {
+                vaultRepository.lockVault(userId)
+            }
+
+            SharedAccountData.Account(
+                userId = account.userId,
+                name = account.name,
+                email = account.email,
+                environmentLabel = account.environment.label,
+                lastSyncTime = lastSyncTime,
+                totpUris = totpUris,
+            )
+        }.let {
```

Comment:
This should be chained:
```kotlin
return allAccounts
    .mapNotNull { account ->
        ...
    }
    .let { SharedAccountData(it) }
```

Reply:
Ah, true. Kinda sad, it means everything inside has to be indented again... but such is life:

eb79ab3e6e833f0de254b6e8fdc2840390426b8e